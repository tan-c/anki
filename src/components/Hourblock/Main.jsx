import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Switch, Route, Redirect } from 'react-router-dom';
import HourblockPageConnected from './HourBlock/Page';
import PlanningPageConnected from './Planning/Page';
import TasksPageConnected from './Task/Page';
import SettingsPageConnected from './Settings/Page';

import FooterConnected from './_layout/Footer';
import LeftAsideConnected from './_layout/LeftAside';
import RightAsideConnected from './_layout/RightAside';

// The Main component renders one of the three provided
// Routes (provided that one matches). Both the /roster
// and /schedule routes will match any pathname that starts
// with /roster or /schedule. The / route will only match
// when the pathname is exactly the string "/"

import './hourblock.scss';

export class MainHourBlock extends React.Component {
  render() {
    const { showMonthlyCalendar } = this.props;

    return (
      <main id="hourblock" data-role="main-hourblock" className={`${window.isMobile && 'mobile'} ${showMonthlyCalendar && 'with-right-calendar'}`}>
        { !window.isMobile
        && (
          <React.Fragment>
            <FooterConnected />
            <LeftAsideConnected />
            {showMonthlyCalendar && <RightAsideConnected />}
          </React.Fragment>
        )
        }

        <Switch>
          <Route exact path="/hourblock" component={HourblockPageConnected} />
          <Route path="/hourblock/planning" component={PlanningPageConnected} />
          <Route path="/hourblock/tasks" component={TasksPageConnected} />
          <Route path="/hourblock/settings" component={SettingsPageConnected} />
        </Switch>
      </main>);
  }
}

MainHourBlock.defaultProps = {
  showMonthlyCalendar: false,
};

MainHourBlock.propTypes = {
  showMonthlyCalendar: PropTypes.bool,
};


function mapStateToProps(state, ownProps) {
  return {
    showMonthlyCalendar: state.ui.getIn(['hourblock', 'hourblockPage', 'showMonthlyCalendar']),
  };
}

export default connect(mapStateToProps)(MainHourBlock);
