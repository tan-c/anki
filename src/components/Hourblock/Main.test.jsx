import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { MainHourBlock } from './Main';

function setup() {
  const props = {
  };

  return shallow(<MainHourBlock {...props} />);
}

describe('MainHourBlock Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("main[data-role='main-hourblock']")).toHaveLength(1);
    });
  });
});
