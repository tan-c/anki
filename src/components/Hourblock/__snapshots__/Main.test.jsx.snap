// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHourBlock Unit Tests when normally set up renders correctly 1`] = `
<main
  className="undefined false"
  data-role="main-hourblock"
  id="hourblock"
>
  <Connect(Footer) />
  <LeftAside />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/planning"
    />
    <Route
      component={[Function]}
      path="/tasks"
    />
    <Route
      component={[Function]}
      path="/settings"
    />
    <Redirect
      from="/*"
      push={false}
      to="/"
    />
  </Switch>
</main>
`;
